pkg/descheduler/strategies/node_taint.go

```diff
@@ -0,0 +1,157 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategies
+
+import (
+	"github.com/kubernetes-incubator/descheduler/cmd/descheduler/app/options"
+	"github.com/kubernetes-incubator/descheduler/pkg/api"
+	"github.com/kubernetes-incubator/descheduler/pkg/descheduler/evictions"
+	podutil "github.com/kubernetes-incubator/descheduler/pkg/descheduler/pod"
+
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+const (
+	TolerationOpExists v1.TolerationOperator = "Exists"
+	TolerationOpEqual  v1.TolerationOperator = "Equal"
+)
+
+// RemovePodsViolatingNodeTaints with elimination strategy
+func RemovePodsViolatingNodeTaints(ds *options.DeschedulerServer, strategy api.DeschedulerStrategy, policyGroupVersion string, nodes []*v1.Node, nodePodCount nodePodEvictedCount) {
+	if !strategy.Enabled {
+		return
+	}
+	deletePodsViolatingNodeTaints(ds.Client, policyGroupVersion, nodes, ds.DryRun, nodePodCount, ds.MaxNoOfPodsToEvictPerNode, ds.EvictLocalStoragePods)
+}
+
+// deletePodsViolatingNodeTaints evicts pods on the node which violate NoSchedule Taints on nodes
+func deletePodsViolatingNodeTaints(client clientset.Interface, policyGroupVersion string, nodes []*v1.Node, dryRun bool, nodePodCount nodePodEvictedCount, maxPodsToEvict int, evictLocalStoragePods bool) int {
+	podsEvicted := 0
+	for _, node := range nodes {
+		glog.V(1).Infof("Processing node: %#v\n", node.Name)
+		pods, err := podutil.ListEvictablePodsOnNode(client, node, evictLocalStoragePods)
+		if err != nil {
+			//no pods evicted as error encountered retrieving evictable Pods
+			return 0
+		}
+		totalPods := len(pods)
+		for i := 0; i < totalPods; i++ {
+			if maxPodsToEvict > 0 && nodePodCount[node]+1 > maxPodsToEvict {
+				break
+			}
+			if !checkPodsSatisfyTolerations(pods[i], node) {
+				success, err := evictions.EvictPod(client, pods[i], policyGroupVersion, dryRun)
+				if !success {
+					glog.Errorf("Error when evicting pod: %#v (%#v)\n", pods[i].Name, err)
+				} else {
+					nodePodCount[node]++
+					glog.V(1).Infof("Evicted pod: %#v (%#v)", pods[i].Name, err)
+				}
+			}
+		}
+		podsEvicted += nodePodCount[node]
+	}
+	return podsEvicted
+}
+
+// checkPodsSatisfyTolerations checks if the node's taints (NoSchedule) are still satisfied by pods' tolerations.
+func checkPodsSatisfyTolerations(pod *v1.Pod, node *v1.Node) bool {
+	tolerations := pod.Spec.Tolerations
+	taints := node.Spec.Taints
+	if len(taints) == 0 {
+		return true
+	}
+	noScheduleTaints := getNoScheduleTaints(taints)
+	allTolerated := allTaintsTolerated(noScheduleTaints, tolerations)
```

Comment:
Don't need to declare a variable here, you could just check `if !allTaintsTolerated(noScheduleTaints, tolerations) {...`